### DISCUSSION TITLE
Configurable and user managed collection list saved filters

### DISCUSSION THREAD

**Discussion by DanRibbens** [payloadcms/payload#3092] [Roadmap]
Filters and search on the collection should be able to be stored in two ways and exposed in the UI to be quickly selected again.

Config based:
1. The collection config would be extended (what do we want to name this property?) adding a new array of objects with:
  - `search` - a string or function that returns a string.
  - `where` - an object or function returns the object to manage the where query.
  - `label` - a string or translated object strings (`{[language: string]: string}`) to name it in the UI.
  - `sort` - an optional string or function returning a string to order the list.
  - `hidden` - a function called with the current `user` to limit which users have the predefined filter.
2. These would be displayed somewhere in the admin UI so that a user can choose from the available choices quickly. Clicking the filter would navigate the user to the list results with the filter applied.
3. If a user changes the `search` or `filters`, the view would no longer appear selected and they could save again as a new user defined view.

User defined:
1. When a user makes changes to the filter options or search within the list view a new control would appear allowing the user to save the current filters to their user preferences.
2. The workflow for saving would let them give it a name.
3. There needs to be a way to delete filters that are no longer used.
4. Optional: we could allow a user to save updates which would overwrite it with new criteria, otherwise just allowing them to save new or delete could be sufficient.

Optional additional feature - Shared user defined:
1. Similar to User defined, saving with this option selected in step 2, would make it available to all other admin users.
2. The config should have a flag to disable this option either as a boolean or a function based on the `user`.
For this to work within preferences, preferences would need a way to be shared across users. Currently the API filters any preferences requests to the current user.

What would this feature be called? It isn't _views_ as that should be reserved for a different layout for the list, like "table" which we have now and we have plans to add more in the future. I'm stuck on names with the word filter, `filters`, `presetFilters`, `userFilters`, `listFilters`, maybe there is a better alternative?

**Comment by jacobsfletch** [comment]
Duplicative of #4193.
